### PR TITLE
Support timeout in Zenoh query

### PR DESCRIPTION
**Pull request by evshary**
The PR is mainly for

* Support timeout for Zenoh query, just like [Rust one](https://github.com/eclipse-zenoh/zenoh/blob/b8dd01d6ce05f851f33497e8d899df308d315afb/examples/examples/z_get.rs#L37)
* ~~Update README format~~

### CODE REVIEW

**Comment by evshary** [comment]
@wyfo Would you mind taking a look at this? Hope this can be included in rc.3

File: src/session.rs [hunk 1]
```diff
@@ -159,6 +160,15 @@ impl _Session {
                 Err(crate::ExtractError::Other(e)) => return Err(e),
                 _ => {}
             }
+            match kwargs.extract_item::<f64>("timeout") {
+                Ok(timeout) => {
+                    builder = builder.timeout(Duration::try_from_secs_f64(timeout).or(Err(
```
Comment (wyfo) [inline src/session.rs:165]:
```suggestion
                    builder = builder.timeout(Duration::try_from_secs_f64(timeout).or_else(|| Err(
```

File: zenoh/session.py [hunk 2]
```diff
@@ -199,6 +199,7 @@ def get(self, selector: IntoSelector, handler: IntoHandler[Reply, Any, Receiver]
         :param target: The queryables that should be target to this query
         :param value: An optional value to attach to this query
         :param attachment: An optional attachment to send along with this query
+        :param timeout: An optional value to the timeout of this query
```
Comment (wyfo) [inline zenoh/session.py:202]: Should add the unit in the description

**Review by wyfo — COMMENTED** [review]
I would prefer to have the documentation update in another PR to make modification tracking easier.

File: src/session.rs (hunk 1 above)
Comment (wyfo) [reply src/session.rs:165]: Also, look at https://github.com/ZettaScaleLabs/zenoh-python/blob/83a193373a94b45ac33694922d3da4bf03080d03/src/session.rs#L245C1-L253C2

File: zenoh/session.py (hunk 2 above)
Comment (evshary) [reply zenoh/session.py]: Make sense

**Comment by evshary** [comment]
> I would prefer to have the documentation update in another PR to make modification tracking easier.

Totally agree. I'll create another PR for that. I shouldn't be too lazy :laughing:

File: src/session.rs [hunk 3]
```diff
@@ -159,6 +160,16 @@ impl _Session {
                 Err(crate::ExtractError::Other(e)) => return Err(e),
                 _ => {}
             }
+            match kwargs.extract_item::<f64>("timeout") {
+                Ok(timeout) => {
+                    builder =
+                        builder.timeout(Duration::try_from_secs_f64(timeout).map_err(|_| {
+                            pyo3::exceptions::PyTypeError::new_err("Wrong timeout value")
```
Comment (wyfo) [inline src/session.rs:167]:
```suggestion
                            pyo3::exceptions::PyValueError::new_err("Wrong timeout value")
```

Comment (wyfo) [reply src/session.rs:167]: Sorry, I was wrong, it should be a `ValueError`

Comment (evshary) [reply src/session.rs]: Make sense

**Comment by wyfo** [comment]
LGTM

**Pull request merged**